pkg/controller/security/security_manager.go

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package security
+
+import (
+	"sync"
+	"time"
+
+	"container/heap"
+
+	istiosecurity "istio.io/istio/pkg/security"
+	"istio.io/istio/pkg/spiffe"
+	"kmesh.net/kmesh/pkg/bpf"
+	"kmesh.net/kmesh/pkg/controller/workload"
+	"kmesh.net/kmesh/pkg/controller/workload/cache"
+	"kmesh.net/kmesh/pkg/logger"
+	"kmesh.net/kmesh/pkg/nets"
+)
+	
+type Cert struct {
+	hostName string
+	exp time.Time
+}
+
+type PriorityQueue []*Cert
+
+var (
+    certsMaps sync.Map
+	ScClient SecretManagerClient
+    log = logger.NewLoggerField("security")
+)
+
+type SecretManagerClient struct {
+	caClient *KmeshCaClient
+
+	// configOptions includes all configurable params for the cache.
+	configOptions *istiosecurity.Options
+
+	// storing certificates
+	secretcache *sync.Map
+
+	//caRootPath string
+	pending	PriorityQueue
+}
+
+const (
+	rootCertPath = "/var/run/secrets/istio/root-cert.pem"
+)
+   
+func (pq PriorityQueue) Len() int { return len(pq) }
+
+func (pq PriorityQueue) Less(i, j int) bool {
+	return pq[i].exp.Before(pq[j].exp)
+}
+
+func (pq PriorityQueue) Swap(i, j int) {
+	pq[i], pq[j] = pq[j], pq[i]
+}
+
+func (pq *PriorityQueue) Push(x interface{}) {
+	item := x.(*Cert)
+	*pq = append(*pq, item)
+}
+
+func (pq *PriorityQueue) Pop() interface{} {
+	old := *pq
+	n := len(old)
+	item := old[n-1]
+	*pq = old[0 : n-1]
+	return item
+}
+
+// NewSecretManagerClient creates a new SecretManagerClient.
+func NewSecretManagerClient() error {
+
+	tlsOpts = &TLSOptions{
+		RootCert:      rootCertPath,
+	}
+
+	options:= NewSecurityOptions()
+	caClient, err := NewKmeshCaClient(options, tlsOpts)
+	if err != nil {
+		return err
+	}
+	
+	pq := make(PriorityQueue, 0)
+	heap.Init(&pq)
+
+	ScClient= SecretManagerClient{
+		caClient:      caClient,
+		configOptions: options,
+		secretcache: &certsMaps,
+		pending:	pq,
+	}
+	go ScClient.refreshExpireCerts()
+	go ScClient.updateCerts()
+	return nil
+}
+ 
+func GetSecretManagerClient() *SecretManagerClient {
+	return &ScClient
+}
+	
+// Automatically check and refresh when the validity period expires
+func (sc *SecretManagerClient) refreshExpireCerts() {
+	var new_certs *istiosecurity.SecretItem
+	var err error
+	var expireTime time.Time
```

Comment:
In go, it is not necessary to define a variable before using it

Reply:
OK